pages/content/amp-dev/documentation/courses/advanced-course/appendix.md

```diff
@@ -70,3 +72,73 @@ Throughout these courses, you will need various images to complete the exercises
 Open [this](https://glitch.com/edit/#!/enshrined-eyebrow) project. Click the “Remix This” button on the top right. This will create a new project that you can edit. You can continue to use this same editor for this and future trainings. Each future tutorial will give you the opportunity to start with a reference version of the solution up to that point.
 
 It’s not necessary to use Glitch to complete these trainings. However, some of the code required to complete the exercises is not discussed in these courses, but is contained in the Glitch samples. If you’d like to use another editor, you may still need to go into the Glitch samples to copy the CSS and server code into your local solution.
+[/filter]
+[filter level="advanced"]
+## What is JSON?
+
+`<amp-list>`, `<amp-bind>` and `<amp-state>` can read JSON data. JSON is short for JavaScript Object Notation. It is a "lightweight data-interchange format."  Even though it is based on a subset of JavaScript, it is language independent, which means that all languages can read and understand it. JSON data are represented, usually, in two ways: either as a collection of key (name) and value pairs, or as an ordered list of values (what we know as a list or array). The JSON syntax looks usually like this:
+
+[sourcecode:json]
+{
+    "people": {
+        "number": 2,
+        "names": ["Alice", "Bob"]
+    }
+}
+[/sourcecode]
+
+If you want a full specification, you can look at the [official documentation](https://www.json.org/).
+
+## What is Deep Merge?
+
+We explained that when `AMP.setState()` is called, `<amp-bind>` will "deep-merge the provided object literal with the current state." The deep-merge functionality will not overwrite anything, but will rather merge the two objects. [In the example]({{g.doc('/content/amp-dev/documentation/components/reference/amp-bind.md', locale=doc.locale).url.path}}#deep-merge-with-ampsetstate) on the `<amp-bind>` documentation, we see that providing only an age for the employee will just update the value of the age of the employee in the state.
+
+[sourcecode:html]
+{% raw %}<button on="tap:AMP.setState({employee: {name: 'John Smith', age: 47, vehicle: 'Car'}})"...></button>
```

Comment:
//cc @morsssss it's better to pretty print bindings and actions to make them easier to understand:

```
<button on="tap:AMP.setState({
  employee: {
    name: 'John Smith', 
    age: 47, vehicle: 'Car'
  }
})". ...>
</button>
```

Reply:
Thanks!

//cc @robocel